zeebe/bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/AdHocSubProcessValidator.java

```diff
@@ -66,31 +64,37 @@ private static void validateTaskDefinition(
       return;
     }
 
-    final ZeebeAdHocImplementationType implementationType =
-        extensionElements.getChildElementsByType(ZeebeAdHoc.class).stream()
+    final ZeebeTaskDefinition taskDefinition =
+        extensionElements.getChildElementsByType(ZeebeTaskDefinition.class).stream()
             .findFirst()
-            .map(ZeebeAdHoc::getImplementationType)
-            .orElse(ZeebeAdHocImplementationType.BPMN);
+            .orElse(null);
 
-    final Collection<ZeebeTaskDefinition> taskDefinitions =
-        extensionElements.getChildElementsByType(ZeebeTaskDefinition.class);
+    if (taskDefinition == null) {
+      return;
+    }
 
-    if (implementationType == ZeebeAdHocImplementationType.BPMN && !taskDefinitions.isEmpty()) {
+    if (!adHocSubProcess.isCancelRemainingInstances()) {
```

Comment:
The attributes defaults to `true`, so we can raise an error if it is not the default value.